trie/bfsIterator.go

```diff
@@ -0,0 +1,33 @@
+package trie
+
+import (
+	"github.com/ElrondNetwork/elrond-go/common"
+)
+
+type bfsIterator struct {
+	*baseIterator
+}
+
+// NewBFSIterator creates a new level order traversal iterator
+func NewBFSIterator(trie common.Trie) (*bfsIterator, error) {
```

Comment:
Why do we've implemented this if will not go to use it anyway? 
Otherwise, I would go for a factory & config value

Reply:
removed the old bfs implementation